starlink_grpc.py

```diff
@@ -395,24 +396,24 @@ class ChannelContext:
     `close()` should be called on the object when it is no longer
     in use.
     """
-    def __init__(self, target=None):
+    def __init__(self, target: str | None = None) -> None:
         self.channel = None
         self.target = "192.168.100.1:9200" if target is None else target
 
-    def get_channel(self):
+    def get_channel(self) -> Tuple[grpc.Channel, bool]:
         reused = True
         if self.channel is None:
             self.channel = grpc.insecure_channel(self.target)
             reused = False
         return self.channel, reused
 
-    def close(self):
+    def close(self) -> None:
         if self.channel is not None:
             self.channel.close()
         self.channel = None
 
 
-def call_with_channel(function, *args, context=None, **kwargs):
+def call_with_channel(function, *args, context: ChannelContext = None, **kwargs):
```

Comment:
I'm no expert on Python type hinting, but it seems inconsistent with below that this does not have a `| None` in the type for context.

Reply:
Actually, now that I'm reading through the PEPs for this, the `|` notation only appears to have just been added in Python 3.10, is that correct? If so, this change would introduce that version requirement on this project, which I would prefer to avoid, if possible.

Reply:
Yep, that's correct
We can `from __future__ import annotations` to rectify this, or just use `Union`
My linters aren't kicking up a fuss about missing `| None`. I thought it was required, but I'll have to check now 😆

Reply:
Or you could use `Optional[ChannelContext]`, I think.
I'm OK with any of those approaches.

Reply:
I like it! I have implemented `Optional` as I think it's a little more clear than `| None`